wiki: Documented flags for Electron based applications (#1302)
YaLTeR/niri: wiki/Application-Issues.md

> ### Electron based applications

Reviewer: ```suggestion
### Electron applications
```

> Electron based applications can work directly on Wayland, but they don't accept it without configuration. Configuration differs between versions. Information was grabbed from the [ArchWiki](https://wiki.archlinux.org/title/Wayland#Electron).

Reviewer: ```suggestion
Electron-based applications can run directly on Wayland, but it's not the default.
```

> For Electron (>28) you need to set environment variable:

Reviewer: ```suggestion
For Electron > 28, you can set an environment variable:
```

```kdl
environment {
    ELECTRON_OZONE_PLATFORM_HINT "auto"
}
```

> For previous version you need to set flags via CLI:

Reviewer: ```suggestion
For previous versions, you need to pass command-line flags to the target application:
```

```

> --enable-features=UseOzonePlatform
> --ozone-platform-hint=auto

Reviewer: ```suggestion
--enable-features=UseOzonePlatform --ozone-platform-hint=auto
```

```

> If application has a [desktop entry](https://specifications.freedesktop.org/menu-spec/latest/menu-add-example.html), you can modify an `Exec` section.

Reviewer: ```suggestion
If the application has a [desktop entry](https://specifications.freedesktop.org/menu-spec/latest/menu-add-example.html), you can put the command-line arguments into the `Exec` section.
```

### Electron based applications

> Electron based application can work directly on Wayland, but don't accept it without configuration. Configuration differs between versions. Infromation was grabbed from [ArchWiki](https://wiki.archlinux.org/title/Wayland#Electron).

Reviewer 2: ```suggestion
Electron based applications can work directly on Wayland, but they don't accept it without configuration. Configuration differs between versions. Information was grabbed from the [ArchWiki](https://wiki.archlinux.org/title/Wayland#Electron).
```

For Electron (>28) you need to set environment variable:
```kdl
environment {

>     ELECTRON_OZONE_PLATFORM_HINT "auto"

Reviewer: Should this say "wayland" instead?

Author: Auto first check wayland, then fallback to x11. I think it is good, if XWayland still present or niri used in combination with other WMs.
https://www.electronjs.org/docs/latest/api/environment-variables#electron_ozone_platform_hint-linux

Reviewer: I see. Makes you wonder why it's still not the default

Reviewer 3: Oh, I didn't know that electron apps respect this variable. I copied system `.desktop` files tom my users home to overwrite the `exec` commands, and append `--ozone-platform-hint=auto` instead. This is much better :heart:

}
```

> For previous version you need to set flags in configuration file `~/.config/electron-flags.conf`:

Reviewer: This file is Arch-specific, so we can't document it here

Author: Oh, I thought it is electron based, but you are right. I will remove those and just paste about command line flags.

Author: Removed.

```
--enable-features=UseOzonePlatform
--ozone-platform-hint=auto
```

### VSCode

If you're having issues with some VSCode hotkeys, try starting `Xwayland` and setting the `DISPLAY=:0` environment variable for VSCode.
That is, still running VSCode with the Wayland backend, but with `DISPLAY` set to a running Xwayland instance.

> Apparently, VSCode currently unconditionally queries the X server for a keymap.

Reviewer: Why did you remove these lines? They apply to VSCode running natively on Wayland.

Author: because if you run VSCode via platform "wayland", then you don't need to start Xwayland and also no issues with hotkeys.

Reviewer: I added this in response to someone literally having this exact issue when running VSCode on Wayland..

Reviewer: I mean, maybe they fixed the problem since? But I wasn't affected by this so idk

Author: Okay, I'll revert

Author: Reverted.

> To run VSCode natively via Wayland you need to put flags in `~/.config/code-flags.conf`:

Reviewer: I suspect this is also Arch-specific

Reviewer 2: i believe so as well https://gitlab.archlinux.org/archlinux/packaging/packages/code/-/blob/main/code.sh?ref_type=heads#L5

Author: Removed.
